cli_tools/common/utils/logging/firelog_entry.go

```diff
@@ -0,0 +1,207 @@
+//  Copyright 2019 Google Inc. All Rights Reserved.
+//
+//  Licensed under the Apache License, Version 2.0 (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+//  Unless required by applicable law or agreed to in writing, software
+//  distributed under the License is distributed on an "AS IS" BASIS,
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+
+package logging
+
+import (
+	"encoding/json"
+	"fmt"
+	"runtime"
+	"time"
+)
+
+// LogRequest is a server-side pre-defined data structure
+type LogRequest struct {
```

Comment:
Will you ever send events for more than a single log source? If so, consider adding the BatchedLogRequest message.

Reply:
My current understanding is that we will send it to one single log source, and it will be routed to 2 sawmill table. Does that sound correct?

Reply:
That's something only your team would know--it sounds right to me if it sounds right to you.